extensions/wikia/ContributionAppreciation/templates/ContributionAppreciation_diffModule.php

```diff
@@ -0,0 +1,6 @@
+<div class="appreciation-diff-module">
+	<span class="appreciation-cta"><?= wfMessage( 'appreciation-cta-text', $username )->escaped() ?></span>
+	<span class="appreciation-feedback"><?= wfMessage( 'appreciation-feedback-text', $username )->escaped() ?></span>
+	<button class="appreciation-button"
+			data-revision="<?= $revision ?>"><?= wfMessage( 'appreciation-text' )->escaped() ?></button>
```

Comment:
This data-revision attribute in this and the historyModule template is vulnerable to XSS because the controller methods can be called directly (i.e. try requesting `/wikia.php?controller=ContributionAppreciation&method=diffModule&revision="><script>alert(document.domain);</script>` in Firefox).